src/main/java/org/springframework/integration/dsl/transaction/TransactionInterceptorBuilder.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.dsl.transaction;
+
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.annotation.Isolation;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.interceptor.DefaultTransactionAttribute;
+import org.springframework.transaction.interceptor.MatchAlwaysTransactionAttributeSource;
+import org.springframework.transaction.interceptor.TransactionAttribute;
+import org.springframework.transaction.interceptor.TransactionInterceptor;
+
+/**
+ * Provides a fluent API to build a transaction interceptor. See
+ * {@link TransactionAttribute} for property meanings; if a {@link TransactionAttribute}
+ * is provided, the individual properties are ignored. If a
+ * {@link PlatformTransactionManager} is not provided, a single instance of
+ * {@link PlatformTransactionManager} will be discovered at runtime; if you have more
+ * than one transaction manager, you must inject the one you want to use here.
+ * @author Gary Russell
+ * @since 1.2
+ *
+ */
+public class TransactionInterceptorBuilder {
+
+	private Propagation propagation = Propagation.REQUIRED;
+
+	private Isolation isolation = Isolation.DEFAULT;
+
+	private int timeout = -1;
+
+	private boolean readOnly = false;
```

Comment:
I'd say we don't need these properties here and just have `DefaultTransactionAttribute final` instance and delegate from the setters.
OTOH, I see that we even can have `final TransactionInterceptor` and use its getters to modify its internals.
WDYT?